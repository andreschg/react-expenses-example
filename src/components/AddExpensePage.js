import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    //props.dispatch(addExpense(expense));
    // This is easier to test
    this.props.startAddExpense(expense);
    // Redirect to the dashboard page.
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
          <h1>Add Expense</h1>
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);