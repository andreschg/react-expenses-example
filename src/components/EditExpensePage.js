import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {

  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  }

  onRemoveButtonClick = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }

  render() {
    return (
      <div>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}/>
        <button onClick={this.onRemoveButtonClick}>Remove</button>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  expense: state.expenses.find((expense) => expense.id === ownProps.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  editExpense: (id, expense) => { dispatch(editExpense(id, expense)); },
  removeExpense: (id) => { dispatch(removeExpense({id})); }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);