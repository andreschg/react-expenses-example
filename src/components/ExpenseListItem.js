import React from 'react';

const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <p>Description: {description}</p>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
  </div>
);

export default ExpenseListItem;