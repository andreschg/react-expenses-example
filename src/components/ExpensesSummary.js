import React from 'react';
import { connect } from 'react-redux';
import visiblesExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = ({ expensesCount, expensesTotal}) =>(
  <div>
    <h1>Viewing {expensesCount} { expensesCount === 1 ? 'expense' : 'expeneses'} totalling {numeral(expensesTotal / 100).format('$0,0.00')}</h1>
  </div>
);

const mapStateToProps = (state) => {
  const expenses = visiblesExpenses(state.expenses, state.filters);
  return {
    expensesCount: expenses.length,
    expensesTotal: expensesTotal(expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);