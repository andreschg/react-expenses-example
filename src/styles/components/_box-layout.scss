.box-layout {
  background-image: url(../images/bg.png);
  height: 100vh;
}