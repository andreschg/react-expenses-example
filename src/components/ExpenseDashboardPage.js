import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <h1>Dashboard</h1>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;