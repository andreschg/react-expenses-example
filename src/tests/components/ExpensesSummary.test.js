import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ExpensesSummary expensesCount={expenses.length} expensesTotal={114195}/>);
});

test('Should render ExpensesSummary component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('Should render Expeneses summary componentes with no data', () => {
  wrapper.setProps({
    expensesCount: 0,
    expensesTotal: 0
  });
  expect(wrapper).toMatchSnapshot();
});