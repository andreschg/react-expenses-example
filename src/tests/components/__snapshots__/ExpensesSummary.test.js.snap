// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expeneses summary componentes with no data 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expeneses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`Should render ExpensesSummary component correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expeneses
     totalling 
    $1,141.95
  </h1>
</div>
`;
